File: typescript/01/star_one.ts
import * as fs from "fs/promises";
import * as path from "path";

(async () => {
    const rawInput = await fs.readFile(path.resolve(__dirname, '../../input/day_01.txt'), "utf-8");
    const arr = rawInput.split("\n");

    const total = arr.reduce((sum, line) => {
        const number_matches = line.match(/[0-9]/g)!
        const num = parseInt(number_matches.at(0) + number_matches!.at(-1)!);  // assume AOC is fair
        return sum + num;
    }, 0);

    console.log(total);
})();

File: python/01/star_one.py
import re
import os

input_file = os.path.join(os.path.dirname(__file__), "../../input/day_01.txt")
re_numbers = re.compile(r"[0-9]")

total = 0
with open(input_file, "r") as file:
    for x in file.readlines():
        nums = re_numbers.findall(x)
        total += int(nums[0] + nums[-1])
print(total)
